### ISSUE TITLE
Running Faster-R-CNN code with no change from git, but getting worse mAP than the reported results

### ISSUE THREAD

**Issue by jwook1004** [tensorpack/tensorpack#780]
Hi,

Thanks for all the hardwork on putting the examples for R-CNNs; this is very cool. 

I am trying to run the Faster-R-CNN code (https://github.com/tensorpack/tensorpack/tree/master/examples/FasterRCNN) and reproduce the results. I installed all the dependencies, set all the data, and used the following command to run training for Faster-R-CNN with the mask option on.
`./train.py --gpu 4,5,6,7 --load COCO-ResNet50-MaskRCNN.npz`

After 280K steps of training, I tried to check mAP by running the following command:
`./train.py --evaluate output.json --load train_log/maskrcnn_0523/model-280000. --gpu 0,1,2,3`

which gives me the following mAP results:
![image](https://user-images.githubusercontent.com/39837223/40853992-ebe6de3e-659d-11e8-9522-a4c9d4398b18.png)

I noticed that this mAP results (31.6/28.6 at IoU(0.5:0.95)) are much worse than the results reported in the website (https://github.com/tensorpack/tensorpack/tree/master/examples/FasterRCNN#results), which is as follows: 
> Backbone | FASTRCNN_BATCH | resolution | schedule | mAP (bbox/segm)
> R50-C4 | 256 | (800, 1333) | 280k | 36.8/32.1
> R50-C4 | 512 | (800, 1333) | 360k | 37.8/33.1

Is there some hyper-parameters I should change to get mAP similar to what is reported? I didn't change "config.py"; e.g., RPN_BATCH_PER_IM = 256 and FASTRCNN_BATCH_PER_IM = 512

From Tensorboard, I can see that mAP is not monotonically increasing over the steps; It's behavior is quite strange, as shown below... (I don't understand why mAP decreases over the steps until step 360K...) 

![image](https://user-images.githubusercontent.com/39837223/40854186-9aecd014-659e-11e8-83bc-6045ce8a7e22.png)

Could you help me understand why I don't get the baseline mAP as reported in the website? Is there any parameters that I need to change further to reproduce the baseline results? 

Thanks for your help,
Jungwook

**Comment by ppwwyyxx** [comment]
You should load a ResNet50 model to train, not a MaskRCNN model. The MaskRCNN model is fully trained already and evaluating that model without training should get you the right score.

**Comment by jwook1004** [comment]
Thanks for the prompt response! Confirmed that the pretrained COCO-ResNet50-MaskRCNN.npz network can achieve mAP of 0.377/0.330 (bbox/segm), which is same as the baseline. Trying "train-from-scratch" with the pretrained ResNet50 model (ImageNet-ResNet50.npz) downloaded from the website (http://models.tensorpack.com/ResNet/). Will update the status.

**Comment by jwook1004** [comment]
Confirmed that train with "ImageNet-ResNet50.npz" pretrained network achieved the baseline as well. Thanks for the help! Closing the issue.
